File: simple_maths/palindrome_string.py



test_cases = int(input())




def palindrome_string(name):
	length = len(name)
	for i in range(length/2):
		if name[i] != name[length-1-i]:
			print("Not a Palindrome")
			return
	print("Palindrome")

for i in range(test_cases):
	name = raw_input()
	palindrome_string(name)

File: simple_maths/palindrome_string.cpp


#include<iostream>


using namespace std;

void palindrome_string(string name)
{
	// sizeof gives the size of the variable
	int length;
	length = name.length();
	for( int i = 0; i< length/2; i++)
	{
		if(name[i] != name[length-1-i])
		{
			cout<<"Not a Palindrome String"<<endl;
			return;
		}
	}
	cout<<"Palindrome String"<<endl;


}

int main()
{

	int test_cases;
	string name;
	cin>>test_cases;

	for(int i = 0; i< test_cases; i++)
	{
		cin>>name;
		palindrome_string(name);
	}

return 0;
}
